src/components/Banner.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from 'react';
+import { ShowData } from '../types';
+import { Typography as Typ } from '@mui/material';
+import SearchInput from './SearchInput';
+
+interface BannerProps {
+    /**
+     * Array of data to randomly choose an image
+     */
+    data: ShowData[] | null;
+    /**
+     * Title that is displayed on the banner
+     */
+    title: string;
+    /**
+     * If true, renders SearchInput component below title
+     */
+    renderSearchInput?: boolean;
+    /**
+     * If true, renders Streamability logo
+     */
+    renderLogo?: boolean;
+}
+
+/**
+ * Image Banner with title and optional search input
+ * @returns {JSX.Element} | Banner
+ */
+const Banner: React.FC<BannerProps> = ({ data, title, renderSearchInput, renderLogo }) => {
```

Comment:
```suggestion
const Banner: React.FC<BannerProps> = ({ data, title, renderSearchInput = false, renderLogo = false }) => {
```
I think it is good to default these to `false` rather than having them undefined.